Add a WithTimescaleDB option to support using the Timescale PG compatible images

ionrock opened the pull request:
This allows support for using Postgres compatible images such as TimescaleDB via:

```
psqldocker.NewContainer(
	u, p, dbname,
	psqldocker.WithRepository("timescale/timescaledb-ha"),
	psqldocker.WithImageTag("pg15-latest"),
)
```

Commit:
Add a repo option to support other pg compatible images

This allows support for using Postgres compatible images such as
TimescaleDB via:

```
psqldocker.NewContainer(
	u, p, dbname,
	psqldocker.WithRepository("timescale/timescaledb-ha"),
	psqldocker.WithImageTag("pg15-latest"),
)
```

adrianbrad commented:
The aim is to maintain a focused approach for this project. maintaining compatibility with select Postgres repositories rather than broad support for any Docker image. In line with this strategy, it's more appropriate to provide options for specific PostgreSQL Docker repositories, with the "postgres" repository being the default one.

For instance, instead of a generic `WithRepository(string)` option, a more specialized `WithTimescaleDB()` option should be provided. This approach aligns better with the project direction.

Also if you intend to update the PR to conform to the above-mentioned ideas, please provide a test case for the new option.

Let me know if I can help in any way!

ionrock commented:
That makes sense to me. I'll update the PR. Thanks for the feedback and taking a look!

Commit:
Specifically support TimescaleDB

This adds the WithTimescaleDB option, which sets the repo to
"timescale/timescaledb" and also sets a valid image tag as the
"alpine" default is not compatible. It only updates the tag when it is
default.

ionrock commented:
@adrianbrad I updated the PR. Let me know what you think!

Commit:
Don't use a receiver name if it is unused

adrianbrad commented:
Excellent 👌. I will bump deps and realease a new version.

ionrock commented:
Fantastic! Thanks so much!
